File: 백준/Silver/14244. 트리 만들기/트리 만들기.py
n,m = map(int,input().split())

now_node=0
count = m-1
for i in range(1,n):
    if m==2:
        print(now_node,i)
        now_node+=1
    else:
        print(now_node,i)
        now_node+=1
        if count:
            now_node-=1
            count-=1

File: 백준/Silver/14244. 트리 만들기/트리 만들기.js
const fs = require('fs');
const input = fs.readFileSync('/dev/stdin', 'utf8').trim();
const [n, m] = input.split(' ').map(Number);

let nowNode = 0;
let count = m - 1;

for (let i = 1; i < n; i++) {
    console.log(nowNode, i);
    nowNode++;

    if (m !== 2 && count > 0) {
        nowNode--;  
        count--;
    }
}
